### PR TITLE
Build: add jsass-vars and completely modularize grunt

### PR DESCRIPTION
**Pull request by arschmitz**
Modular javascript object based compilation of scss varibles

### CODE REVIEW

File: tasks/variables.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+		var jsass = require( "jsass-vars" );
+		var path = require( "path" );
+		var varPath = path.join( process.cwd(), "scss/variables/*" );
+
+		jsass( varPath, {
+			"dest": path.join( process.cwd(), "dist" ),
+			"name": "chassis"
+		} );
+	} )
+};
```
Comment (arthurvr) [inline tasks/variables.js]: missing newline

File: tasks/options/watch.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+		}
+	},
+	svg: {
+		files: [ "svg-source/**/*.svg" ],
+		tasks: [ "svg" ],
+		options: {
+			spawn: false
+		}
+	}
+}
```
Comment (arthurvr) [inline tasks/options/watch.js]: missing newline

File: tasks/options/svgstore.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+				viewBox: "0 0 24 24",
+				xmlns: "http://www.w3.org/2000/svg"
+			},
+			cleanup: [ "style", "fill", "id" ]
+		},
+		files: {
+			"icons/icons.svg": [ "icons/svg-min/*.svg" ]
+		}
+	}
+}
```
Comment (arthurvr) [inline tasks/options/svgstore.js]: missing newline

File: tasks/options/watch.js (hunk 2 above)
Comment (arthurvr) [inline tasks/options/watch.js]: and a semicolon

File: tasks/options/svgstore.js (hunk 3 above)
Comment (arthurvr) [inline tasks/options/svgstore.js]: plus a semicolon

File: tasks/options/sass.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+			sourceMap: true,
+
+			// This actually does nested until libsass updates to support expanded
+			outputStyle: "expanded"
+		},
+		files: {
+			"dist/css/chassis.css": "scss/style.scss"
+		}
+	}
+};
```
Comment (arthurvr) [inline tasks/options/sass.js]: misses a newline

File: tasks/options/jscs.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+	all: [ "*.js", "performance/**/*.js" ]
+};
```
Comment (arthurvr) [inline tasks/options/jscs.js]: newline :)

File: tasks/options/jshint.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	files: [ "*.js", "performance/**/*.js" ],
+	options: {
+		jshintrc: ".jshintrc"
+	}
+};
```
Comment (arthurvr) [inline tasks/options/jshint.js]: newline :)

File: tasks/variables.js (hunk 1 above)
Comment (sfrisk) [inline tasks/variables.js]: man, you waste like no time.

File: scss/variables/index.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+var fs = require( "fs" );
+var path = require( "path" );
+var varPath = path.join( process.cwd(), "scss/variables" );
+var chassis;
+
+fs.readdirSync( varPath ).forEach( function( file ) {
+	chassis = require( path.join( process.cwd(), "scss/variables" ) + "/" + file );
+} );
+
+module.exports = chassis;
```
Comment (arthurvr) [inline scss/variables/index.js]: misses a final newline

File: scss/variables/chassis.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+// This is just boiuler plate dont worry about it
+// This just supports loading in any enviroment
+(function ( root, factory ) {
+    if ( typeof define === "function" && define.amd ) {
```
Comment (arthurvr) [inline scss/variables/chassis.js]: 4 spaces indentation?

File: scss/variables/chassis.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+        define( [], factory );
+    } else if ( typeof exports === "object" ) {
+        module.exports = factory();
+    } else {
+        root.chassis = factory();
+  }
+}( this, function () {
+return {};
+} ) );
```
Comment (arthurvr) [inline scss/variables/chassis.js]: missing newline

File: tasks/options/csslint.js [hunk 10]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+	src: [ "dist/css/*.css" ]
+};
```
Comment (arthurvr) [inline tasks/options/csslint.js]: missing newline

File: scss/variables/template.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+// This is just boiuler plate dont worry about it
+// This just supports loading in any enviroment
+(function ( root, factory ) {
+    if ( typeof define === "function" && define.amd ) {
+        define( [], factory );
+    } else if ( typeof exports === "object" ) {
+        module.exports = factory();
+    } else {
+        root.chassis = factory();
+  }
```
Comment (arthurvr) [inline scss/variables/template.js]: indentation

File: tasks/variables.js (hunk 1 above)
Comment (arthurvr) [inline tasks/variables.js]: lol :p

**Comment by arthurvr** [comment]
That's it for the time being. Love this. Definitely an improvement.

File: package.json [hunk 12]
```diff
@@ -40,15 +40,12 @@
     "start": "grunt watch",
     "test": "grunt"
   },
-  "dependencies": {
-    "normalize.css": "3.0.3"
-  },
+  "dependencies": {},
```
Comment (arthurvr) [inline package.json]: there's not much of a deal with including this when it's empty.

File: scss/atoms/typography/_typography.scss [hunk 13]
```diff
@@ -4,123 +4,4 @@
...
-
-	@media ( max-width: 800px ) {
-		font-size: 16px;
-	}
-	@media ( max-width: 500px ) {
-		font-size: 14px;
-	}
-}
-
-h1 {
```
Comment (sfrisk) [inline scss/atoms/typography/_typography.scss]: Why did you delete all of this?

Comment (arschmitz) [inline scss/atoms/typography/_typography.scss]: it shouldnt be there is some other whole files deleted that should not be as well not sure what happened there ill fix this and @arthurvr comments

File: scss/variables/colors.js [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+	},
+	"chassis-blue": {
+		"value": "#4fc0c8",
+		"name": "Chassis Blue"
+	},
+	"chassis-yellow": {
+		"value": "#fadf51",
+		"name": "Chassis Yellow"
+	}
+}
```
Comment (arthurvr) [inline scss/variables/colors.js]: Strange that JSHint didn't complain about this; missing semicolon.

File: scss/variables/colors.js [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+	"background": colors[ "chassis-gray-dark" ],
+	"font": colors[ "chassis-gray-dark" ],
+	"link": colors[ "chassis-blue-dark" ],
+	"button": colors[ "chassis-blue" ],
+	"buttonText": colors[ "chassis-gray-light" ],
+	"blockquoteBorder": {
+		name: "Chassis - Blockquote color",
+		value: "#eee"
+	}
+}
```
Comment (arthurvr) [inline scss/variables/colors.js]: missing semicolon

File: scss/variables/colors.js (hunk 14 above)
Comment (arschmitz) [inline scss/variables/colors.js]: needed to add this new folder to jscs and jshint done now

File: performance/frameworks/bootstrap.js [hunk 16]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-	css: [ "//maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap.min.css" ],
+	css: [ "//maxcdn.bootstrapcdn.com/bootstrap/3.3.1/css/bootstrap.min.css" ],
```
Comment (arthurvr) [inline performance/frameworks/bootstrap.js]: Hmmm... Are we downgrading?

File: scss/variables/colors.js (hunk 14 above)
Comment (arthurvr) [inline scss/variables/colors.js]: ah! Awesome!

File: tasks/options/autoprefixer.js [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+				"safari >= 5.1",
+				"ios >= 6.1",
+				"android 2.3",
+				"android >= 4",
+				"ie >= 8"
+			]
+		},
+		src: "dist/css/*.css"
+	}
+};
```
Comment (arthurvr) [inline tasks/options/autoprefixer.js]: missing newline

File: tasks/options/csscomb.js [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+			"dist/css/chassis.css": "dist/css/chassis.css"
+		}
+	},
+	scss: {
+		files: [ {
+			expand: true,
+			src: [ "scss/**/*.scss" ]
+		} ]
+	}
+};
```
Comment (arthurvr) [inline tasks/options/csscomb.js]: missing newline

File: package.json [hunk 19]
```diff
@@ -59,14 +58,13 @@
     "grunt-csscomb": "3.0.0",
     "grunt-git-authors": "2.0.0",
     "grunt-html": "1.6.0",
-    "grunt-htmllint": "0.2.7",
-    "grunt-jscs": "0.6.2",
-    "grunt-npmcopy": "0.1.0",
-    "grunt-sass": "0.18.1",
+    "grunt-jscs": "1.8.0",
+    "grunt-sass": "0.17.0",
```
Comment (sfrisk) [inline package.json]: Why was this switched to an older version of grunt-sass?
